tokio-console/src/view/durations.rs

```diff
@@ -0,0 +1,104 @@
+use std::cmp;
+
+use tui::{
+    layout::{self},
+    widgets::Widget,
+};
+
+use crate::{
+    state::histogram::DurationHistogram,
+    view::{self, mini_histogram::MiniHistogram, percentiles::Percentiles},
+};
+
+// This is calculated so that a legend like the below generally fits:
+// │0647.17µs  909.31µs │
+const MIN_HISTOGRAM_BLOCK_WIDTH: u16 = 22;
+
+/// This is a tui-rs widget to visualize durations as a list of percentiles
+/// and if possible, a mini-histogram too.
+///
+/// This widget wraps the [`Percentiles`] and [`MiniHistogram`] widgets which
+/// are displayed side by side. The mini-histogram will only be displayed if
+///   a) UTF-8 support is enabled via [`Styles`]
+///   b) There is at least a minimum width left after drawing the percentiles
+///
+/// [`Styles`]: crate::view::Styles
+pub(crate) struct Durations<'a> {
+    /// Widget style
+    styles: &'a view::Styles,
+    /// The histogram data to render
+    histogram: Option<&'a DurationHistogram>,
+    /// Title for percentiles block
+    percentiles_title: &'a str,
+    /// Title for histogram sparkline block
+    histogram_title: &'a str,
+}
+
+impl<'a> Widget for Durations<'a> {
+    fn render(self, area: tui::layout::Rect, buf: &mut tui::buffer::Buffer) {
+        // Only split the durations area in half if we're also drawing a
+        // sparkline. We require UTF-8 to draw the sparkline and also enough width.
+        let (percentiles_area, histogram_area) = if self.styles.utf8 {
+            let percentiles_width = cmp::max(self.percentiles_title.len() as u16, 13_u16) + 2;
+
+            // If there isn't enough width left after drawing the percentiles
+            // then we won't draw the sparkline at all.
+            if area.width < percentiles_width + MIN_HISTOGRAM_BLOCK_WIDTH {
+                (area, None)
+            } else {
+                let areas = layout::Layout::default()
+                    .direction(layout::Direction::Horizontal)
+                    .constraints(
+                        [
+                            layout::Constraint::Length(percentiles_width),
+                            layout::Constraint::Min(50),
```

Comment:
why is the constraint here `Min(50)` instead of `Min(MIN_HISTOGRAM_BLOCK_WIDTH)`?